pandas/core/reshape/concat.py

```diff
@@ -436,6 +446,8 @@ def __init__(
                     # to line up
                     if self._is_frame and axis == 1:
                         name = 0
+                    # mypy needs to know sample is not an NDFrame
+                    assert isinstance(sample, (ABCSeries, ABCDataFrame))
```

Comment:
I think safe to use as cast here which is a no-op instead of a runtime 'slow' isinstance check on ABCs.

We moreless have a guarantee that we don't have an instance of NDFrame or a subclass that is not a Series or a DataFrame

Reply:
Done.